ads/inabox/inabox-messaging-host.js

```diff
@@ -125,10 +125,19 @@ export class InaboxMessagingHost {
     const iframe =
         this.getFrameElement_(message.source, request['sentinel']);
     if (!iframe) {
-      dev().info(TAG, 'Ignored message from untrusted iframe:', message);
+      dev().info(TAG, 'Ignored message from unmeasurable iframe:', message);
       return false;
     }
 
+    // if message comes from an unregistered frame, allow anyway if it's only
+    // getting viewability
+    if (!this.iframeMap_[request['sentinel']].iframe) {
```

Comment:
I'm not convinced this works: after running `getFrameElement_` above either:

* `getFrameElement_` returned `null`, `!iframe` was `true`, we returned `false` above
* `!this.iframeMap_[request['sentinel']].iframe` will be false and this code block will never run

Have you tested this locally?

Reply:
getFrameElement_ does not return the iframe, it returns the top most measurable frame (measurableFrame), so the two conditions are not mutually exclusive. I had tested this locally and it works on my end.

Reply:
I'm still not seeing it, but maybe I'm misreading?

Case 1:

* `getFrameElement_` returns `null`
* `iframeMap_[sentinel]` is unset
* `processMessage` will log `Ignored message from unmeasurable iframe` and return false

Case 2:

* `getFrameElement_` returns non-`null`
* `iframeMap_[sentinel]` will be set, as will `iframeMap_[sentinel].iframe`
* `!this.iframeMap_[request['sentinel']].iframe` will be false
* the `if (request['type'] != MessageType.SEND_POSITIONS) {...}` block won't ever run

Are you getting `Ignored message from untrusted iframe` in local testing?